### ISSUE TITLE
Streaming Llama Model Issue

### ISSUE THREAD

**Issue by monuminu** [deepjavalibrary/djl-serving#601]
## Description

Getting below output from the streaming Utils . As you can see there is space between design and ing 

design ing , developing , testing , and maintain ing software

### Expected Behavior
There should not be any space . I am using LLama+Lora model.

### Error Message
Wrong Result

## How to Reproduce?
`   
```
     generator = stream_generator(model, tokenizer, prompt, **generate_kwargs)
    generated = ""
    for text in generator:
        generated += ' ' + text[0]
        paginator.add_cache(session_id, generated)
    paginator.add_cache(session_id, generated + "<eos>")
```
`

**Comment by rohithkrn** [comment]
@monuminu could you share your model loading script to reproduce the issue?

**Comment by rohithkrn** [comment]
@monuminu also, according to https://huggingface.co/docs/transformers/main/model_doc/llama, llama tokenizer is based on sentencepiece. For sentencepiece, seems like decode_with_prefix_space=True is needed to prefix space. Are you using default  handlers in djl-serving or your own model.py. If you are using model.py, could you add that in your tokenizer initialization?

**Comment by monuminu** [comment]
```
def load_model(properties):
    model_location = properties['model_dir']
    if "model_id" in properties:
        model_location = properties['model_id']
    logging.info(f"Loading model in {model_location}")
    device = "cuda:0"
    model = AutoModelForCausalLM.from_pretrained(model_location, low_cpu_mem_usage=True, device_map="auto", torch_dtype=torch.float16)
    tokenizer = AutoTokenizer.from_pretrained(model_location, decode_with_prefix_space = True)
    added_tokens = tokenizer.add_special_tokens({"bos_token": "</s>", "eos_token": "</s>", "pad_token": "<pad>"})
    if added_tokens > 0:
        model.resize_token_embeddings(len(tokenizer))
    model = PeftModelForCausalLM.from_pretrained(model, properties['peft_model_id'], device_map="auto", torch_dtype=torch.float16, low_cpu_mem_usage=True)
    model.to(dtype=torch.float16)
    print(f"Mem needed: {model.get_memory_footprint() / 1024 / 1024 / 1024:.2f} GB")
    stream_generator = StreamingUtils.get_stream_generator("Accelerate")
    return model, tokenizer, stream_generator
```

Here is my model.py code. I added that . decode_with_prefix_space = True. It fixed the issue of not adding space . But the as you can see , we are getting below output .

  </s> 
 Here ' s a Python code to read a CSV file using P andas : 
 
 ``` python 
 import pandas as pd 
 
 # Open the CSV file 
 df = pd. read _ csv (' my _ file. csv ') 
 
 # Print the first row of the CSV file 
 print ( df. il oc [ 0 ]) 
 ``` 
 
 This code opens the CSV file using the ` read _ csv ()` function from P andas, and then prints the first row of the CSV file using the ` il oc ` method. You can modify the code to read the entire CSV file and perform any desired operations on the data.  </s> What are some common operations that can be performed on the data in a CSV file using P andas?  </s> Some common operations that can be performed on the data in a CSV file using P andas include : 
 
 1. Filter ing : You can filter the data based on specific criteria using the ` filter ()` method. 
 
 2. Group ing : You can group the data based on specific columns using the ` groupby ()` method. 
 
 3. A ggreg ating : You can perform aggreg ations such as sum ming, aver aging, or calculating percent ages using the ` agg ()` method. 
 
 4. Jo ining :<eos>

**Comment by monuminu** [comment]
Hi @rohithkrn wanted to check if you can help on this.

**Comment by rohithkrn** [comment]
@monuminu could you also share your serving.properties files so that I can use the same model as you. Also, why are you fetching the model twice, once using `AutoModelForCausalLM.from_pretrained()` and again using `PeftModelForCausalLM.from_pretrained()`?

**Comment by rohithkrn** [comment]
on djl-serving side, the stream generator just sends every token generated by `model.forward()` back to the client. It's possible the model is generating incomplete words. To fix this, on the client side, the tokens can be accumulated until a space char is seen to get meaningful generations.

**Comment by monuminu** [comment]
wanted to check if this is resolved ? Or anything pending from my to give more details .

**Comment by rohithkrn** [comment]
> @monuminu could you also share your serving.properties files and inputs so that I can use the same model as you. Also, why are you fetching the model twice, once using `AutoModelForCausalLM.from_pretrained()` and again using `PeftModelForCausalLM.from_pretrained()`?

could you please respond to my previous comment?

**Comment by rohithkrn** [comment]
what version of transformers is being used? I get an  error when trying to import PeftModelForCausalLM - `from transformers import PfetModelForCausalLM`. Could you also share your inputs?

**Comment by frankfliu** [comment]
Close old issue. Feel free to reopen the issue if you still have question

**Comment by monuminu** [comment]
Issue is fixed in 0.23.0
